### PR TITLE
New term entry for Java Math.subtractExact()

### LINKED ISSUE #916: [Term Entry] Java Math Methods: `.subtractExact()`
### Reason/inspiration (optional)

Under `java/concepts/math-methods/terms/`, create a new term entry for the `Math.subtractExact()` Java method.

- In the `terms/` folder, make a new folder called `subtract-exact`
- Inside the `subtract-exact` folder, create a new markdown file called `subtract-exact.md`.
- Inside the file, be sure to include a "Syntax" section with a \```pseudo code snippet.
- Also make sure to include an "Example" section with a \```java code snippet.

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] I agree to follow Codecademy Doc's Code of Conduct

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by shivanshugour45**
### Description

This file contains the explanation of '.subtractExact()' method present in java.lang.Math class.

Closes #916 

### Type of Change

- Adding a new entry

### Checklist

- [v] All writings are my own.
- [v] My entry follows the Codecademy Docs style guide.
- [v] My changes generate no new warnings.
- [v] I have performed a self-review of my own writing and code.
- [v] I have checked my entry and corrected any misspellings.
- [v] I have made corresponding changes to the documentation if needed.
- [v] I have confirmed my changes are not being pushed from my forked `main` branch.
- [v] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [v] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+The **`Math.subtractExact()`** method returns the difference of the arguments.
+
+## Syntax
+
+```pseudo
+Math.subtractExact(int num)
+```
+
+Datatype of the value returned by '.subtractExact()' method depends upon the arguments passed.
```
Comment (SSwiniarski) [inline content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md:24]:
```suggestion
Datatype of the value returned by '.subtractExact()' method depends upon the arguments passed:
```

File: content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+- If both the arguments are of type int then the return type will be int.
```
Comment (SSwiniarski) [inline content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md:26]:
```suggestion
- If both the arguments are of type `int` then the return type will be `int`.
```

File: content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+- If both the arguments are of type long then the return type will be long.
```
Comment (SSwiniarski) [inline content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md:27]:
```suggestion
- If both the arguments are of type `long` then the return type will be `long`.
```

File: content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+- If one of the arguments is of long type then the return will be long.
```
Comment (SSwiniarski) [inline content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md:28]:
```suggestion
- If one of the arguments is of `long` type then the return will be `long`.
```

File: content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+## Example
+
+The following example showcases the `.subtractExact()` method being applied to two variables:
```
Comment (SSwiniarski) [inline content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md:32]:
```suggestion
The following example demonstrates the `.subtractExact()` method being applied to two variables:
```

File: content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`Math.subtractExact()`** method returns the difference of the arguments.
+
+## Syntax
+
+```pseudo
+Math.subtractExact(int num)
```
Comment (SSwiniarski) [inline content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md:21]:
```suggestion
Math.subtractExact(x, y)
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@shivanshugour45, I did a first review and left a couple of comments.

**Review by shivanshugour45 — COMMENTED** [review]
Suggested changes are done.

**Review by SSwiniarski — APPROVED** [review]
@shivanshugour45 Looks good for a second review.

File: content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  - 'Arithmetic'
+  - 'Functions'
+  - 'Methods'
+  - 'Numbers'
+CatalogContent:
+  - 'learn-java'
+  - 'paths/computer-science'
+---
+
+The **`Math.subtractExact()`** method returns the difference of the arguments.
```
Comment (HishamT) [inline content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md:16]: Add an explanation of the difference or benefit of using this function as opposed to the subtraction operator (-).

File: content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+
+## Syntax
+
+```pseudo
+Math.subtractExact(x, y)
+```
+
+Datatype of the value returned by '.subtractExact()' method depends upon the arguments passed:
```
Comment (HishamT) [inline content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md:24]: "~Datatype~ The type of the value returned..."

File: content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+- If both the arguments are of type `int` then the return type will be `int`.
+- If both the arguments are of type `long` then the return type will be `long`.
+- If one of the arguments is of `long` type then the return will be `long`.
+
```
Comment (HishamT) [inline content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md:29]: Place commas in an if-then clause:

- If both the arguments are of type `int`, then the return type will be `int`.
- If both the arguments are of type `long`,  then the return type will be `long`.
- If one of the arguments is of type `long`, ~type~ then the return type will be `long`.

File: content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+## Example
+
+The following example demonstrates the `.subtractExact()` method being applied to two variables:
```
Comment (HishamT) [inline content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md:32]: "The following example demonstrates the `.subtractExact()` method: ~being applied to two variables:~"

**Review by HishamT — CHANGES_REQUESTED** [review]
Hey @shivanshugour45, I finished reviewing your entry and left some comments.

File: content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md (hunk 7 above)
Comment (shivanshugour45) [reply content/java/concepts/math-methods/terms/subtract-exact/subtract-exact.md:16]: I added an explanation to the best of my knowledge. Please suggest improvements.

**Comment by shivanshugour45** [comment]
Hi @HishamT, I've made the suggested changes.

**Comment by shivanshugour45** [comment]
@HishamT, the requested change is done.

**Review by HishamT — APPROVED** [review]
LGTM!

**Comment by yangc95** [comment]
@shivanshugour45 Live link: https://www.codecademy.com/resources/docs/java/math-methods/subtractExact

Screen Shot 2022-10-21 at 9 06 09 PM
You're currently anonymous, but if you would like to make yourself public, make sure to link your GitHub to your Codecademy profile.

cc: @HishamT @SSwiniarski

**Comment by Dusch4593** [comment]
Hey @shivanshugour45! Thank you so much for your contribution to Docs! 🚀 

I was wondering if you were up for opening a new PR with edits for this new entry? I had a few suggestions:

- To improve consistency, can we change `Math.subtractExact()` to just `.subtractExact()` on lines # 18 and # 20.
- Also on lines # 18 and # 20, I would also rephrase as "[T]he `.subtractExact()` method" to maintain consistency.
- On line # 28, can we make the following edit? (emphasis mine)
   > The type of the value returned by _the_ `.subtractExact()` method depends upon the arguments passed:
- On line # 32, can we rephrase the text to something like this?
   > \- If one of the arguments is of type `long`, then the return type will be `long`.

**Comment by shivanshugour45** [comment]
Hi @Dusch4593, Sure will do the changes and create the PR.

**Pull request merged**